### PR TITLE
store `GEOMETRY` types as `BLOB`s

### PR DESCRIPTION
**Pull request by JCOR11599**
To get around the 65k limit on our fields, we should store `GEOMETRY` types as `BLOB`s.
Additionally, MySQL stores these as out of band BLOBs.

This will also probably worsen the performance of `GEOMETRY` types as it just takes longer to reference.

Fixes https://github.com/dolthub/dolt/issues/6927

### CODE REVIEW

File: go/store/val/tuple_descriptor.go [hunk 1]
```diff
@@ -439,6 +439,11 @@ func (td TupleDesc) GetGeometry(i int, tup Tuple) (v []byte, ok bool) {
 	return
 }
 
+func (td TupleDesc) GetGeometryAddr(i int, tup Tuple) (hash.Hash, bool) {
+	td.expectEncoding(i, GeometryEnc)
```
Comment (zachmu) [inline go/store/val/tuple_descriptor.go:443]: I'm a little bothered by the fact that there isn't a separate encoding for this

**Review by zachmu — COMMENTED** [review]
Mostly looks good, like the backwards compatibility

But I think we need a new encoding type if we're going to support both encodings

File: go/libraries/doltcore/sqle/index/prolly_fields.go [hunk 2]
```diff
@@ -91,12 +91,27 @@ func GetField(ctx context.Context, td val.TupleDesc, i int, tup val.Tuple, ns tr
 			err = json.Unmarshal(buf, &doc.Val)
 			v = doc
 		}
+	// TODO: eventually remove this, and only read GeomAddrEnc
 	case val.GeometryEnc:
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/index/prolly_fields.go:95]: Are any tests going to hit this logic to see if it works?

**Review by zachmu — APPROVED** [review]
LGTM, just make sure there are tests that exercise the backward compatibility logic. You may need to manually construct tuples with PutGeometry, because everything on the SQL path is just going to use the new encoding.

File: go/libraries/doltcore/sqle/index/prolly_fields.go (hunk 2 above)
Comment (JCOR11599) [reply go/libraries/doltcore/sqle/index/prolly_fields.go:95]: added some in prolly_field_tests.go

**Pull request merged**